hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -166,153 +164,58 @@ protected void doRefresh() {
-
-      } catch (CommitFailedException | CommitStateUnknownException e) {
-        throw e;
-
-      } catch (Throwable e) {
-        if (e.getMessage()
-            .contains(
-                "The table has been modified. The parameter value for key '"
-                    + HiveTableOperations.METADATA_LOCATION_PROP
-                    + "' is")) {
-          throw new CommitFailedException(
-              e, "The table %s.%s has been modified concurrently", database, tableName);
-        }
-
-        if (e.getMessage() != null
-            && e.getMessage().contains("Table/View 'HIVE_LOCKS' does not exist")) {
-          throw new RuntimeException(
-              "Failed to acquire locks from metastore because the underlying metastore "
-                  + "table 'HIVE_LOCKS' does not exist. This can occur when using an embedded metastore which does not "
-                  + "support transactions. To fix this use an alternative metastore.",
-              e);
-        }
-
-        LOG.error(
-            "Cannot tell if commit to {}.{} succeeded, attempting to reconnect and check.",
-            database,
-            tableName,
-            e);
-        commitStatus = checkCommitStatus(newMetadataLocation, metadata);
-        switch (commitStatus) {
-          case SUCCESS:
-            break;
-          case FAILURE:
-            throw e;
-          case UNKNOWN:
-            throw new CommitStateUnknownException(e);
-        }
-      }
-    } catch (TException e) {
-      throw new RuntimeException(
-          String.format("Metastore operation failed for %s.%s", database, tableName), e);
+  @Override
+  public BaseMetastoreTableOperations.CommitStatus validateNewLocationAndReturnCommitStatus(
+      TableMetadata metadata, String newMetadataLocation) {
+    return MetastoreOperationsUtil.checkCommitStatus(
+        fullName,
+        newMetadataLocation,
+        metadata.properties(),
+        this::calculateCommitStatusWithUpdatedLocation);
+  }
 
-    } catch (InterruptedException e) {
-      Thread.currentThread().interrupt();
-      throw new RuntimeException("Interrupted during commit", e);
+  @Override
+  public StorageDescriptor storageDescriptor(TableMetadata metadata, boolean hiveEngineEnabled) {
+    return HiveOperationsBase.storageDescriptor(
+        metadata.schema(), metadata.location(), hiveEngineEnabled);
+  }
 
-    } catch (LockException e) {
-      throw new CommitFailedException(e);
+  @Override
+  public Set<String> obsoleteProps(TableMetadata base, TableMetadata metadata) {
+    Set<String> obsoleteProps = Sets.newHashSet();
+    if (base != null) {
+      obsoleteProps =
+          base.properties().keySet().stream()
+              .filter(key -> !metadata.properties().containsKey(key))
+              .collect(Collectors.toSet());
+    }
 
-    } finally {
-      cleanupMetadataAndUnlock(commitStatus, newMetadataLocation, lock);
+    if (!conf.getBoolean(ConfigProperties.KEEP_HIVE_STATS, false)) {
```

Comment:
All the obsolete props are being merged together now. Earlier It was being managed by 
[removedProps](https://github.com/apache/iceberg/blob/53a1c8671dd1b9b93f4a857230008c812d79ddbf/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L217) 
and [hiveStatCheck](https://github.com/apache/iceberg/blob/53a1c8671dd1b9b93f4a857230008c812d79ddbf/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L232)